### ISSUE TITLE
i write this code for angular json schema form

### ISSUE THREAD

**Issue by varshildoshi** [dschnelldavis/angular2-json-schema-form#227]
{
                  "type" : "button",
                  "title" : "save",
                  onClick : "exampleOnSubmitFn($event)"
}

but does not work properly

**Comment by Alex130** [comment]
Hi,
can you provide more information about your form (schema, data, layout)?
As a guess, you probably want this for a submit button :
`"type": "submit",
  "style": "btn-info",
  "title" : "save",
   "onClick" : exampleOnSubmitFn($event)`

**Comment by varshildoshi** [comment]
yes,but i want to call my own method through submit button,but its not working..

**Comment by varshildoshi** [comment]
i take one json for generate layout using typescript & angular 4,so if i create my own button for reset,submit,next & save,so need my own function for event,but in my json function is not work proprly...so give me solution..

**Comment by varshildoshi** [comment]
i got the solution...

**Comment by varshildoshi** [comment]
but one more thing,
this is my birthdate json
"birthdate" : {
                "type" : "date"
            },
this is my layout json for birthdate
{
                "key" : "birthdate",
                "type" : "date",
                "flex" : "3 3 100px"
            },
but it gives me wrong data for date,ex: if i select 3/4/2003 it returns 2/4/2003,what can i do for that??

**Comment by varshildoshi** [comment]
if i take type="string" & layout type="date" its not working..

**Comment by varshildoshi** [comment]
& i use angular material design for that..

**Comment by Alex130** [comment]
> "but it gives me wrong data for date,ex: if i select 3/4/2003 it returns 2/4/2003,what can i do for that??"

What timezone are you using? it is related to #191 . 
The bug is not fixed but I wrote an override for my own project. Feel free to use it.
[datePickerOverride.zip](https://github.com/dschnelldavis/angular2-json-schema-form/files/1797402/datePickerOverride.zip)
I might submit a pull request to do things cleanly when I have the time.

In app.modules.ts :
```javascript
{ provide: MAT_DATE_LOCALE, useValue: 'fr-FR' },
    { provide: MAT_DATE_FORMATS, useValue: MOMENT_DATE_FORMATS},
    { provide: DateAdapter, useClass : MomentDateAdapter }
```
and 

```javascript
export class AppModule {
  constructor(wl: WidgetLibraryService) {
    wl.registerWidget('date', NewMaterialDatepickerComponent);
  }
}
```
Replace fr by your desired language
The code was for my usage only, with Moments.js, and not fully tested, but I hope it helps

**Comment by Alex130** [comment]
Btw, using your own function on Submit is done in the template's json-schema-form tag :
```html
<json-schema-form framework="material-design" [layout]="layout" loadExternalAssets="false" [schema]="userScheme" [(ngModel)]="currentUser" (onSubmit)="submit($event)"></json-schema-form>
```
Note the (onSubmit) binding. The event Object contains your validated and submitted form.

**Comment by varshildoshi** [comment]
i am not sure,so plz send me full code with output for datapicker..
plz...

**Comment by varshildoshi** [comment]
and you can share related link for datepicker...for better understanding

**Comment by Alex130** [comment]
Hi, 
[Material Datepicker](https://material.angular.io/components/datepicker/overview#choosing-a-date-implementation-and-date-format-settings) this is the documentation for Material Datepicker. Note the DateAdapter, which is not configurable in angular2-json-schema-form in its current state.
After modifying your app.modules [as above](https://github.com/dschnelldavis/angular2-json-schema-form/issues/227#issuecomment-371848533), the code in my previous [comment ](https://github.com/dschnelldavis/angular2-json-schema-form/issues/227#issuecomment-371849436) is all you need to use the overriden class automatically, and the event in your submit function should contain the correct date (with your locale and the correct date for your timezone).
You can keep your schema and layout as it is in [this post](https://github.com/dschnelldavis/angular2-json-schema-form/issues/227#issuecomment-371794977) .

**Comment by varshildoshi** [comment]
Thanks..Alex130

**Comment by varshildoshi** [comment]
one more thing..Alex130
I want to use tinymce in my project,so if u have any idea about it..plz tell me...in angular 4,angular material

**Comment by Alex130** [comment]
Hi, it's not implemented yet, you would need to write your own widget, you
can see the developer of the module writes about it in the future versions
section on the main github page.
If you do, please contribute to it if you can

**Comment by varshildoshi** [comment]
ok..sure..
but do you know how to define & use my own css in json using angular material

**Comment by Alex130** [comment]
I did a pull request about this a while ago : #221
Some attributes are already bound and you can use them, I added some of the missing ones, check it out.

**Comment by varshildoshi** [comment]
hi Alex130
i want to help from you..
i want to create Drag & Drop in angular 4 using typescript,but some problem occures..
when i drop any text view generate text view in dropzone..it is using angular material..
if you know about that plz tell me..i am use jquery for drag & drop with angular material..so plz help me..
